### PR TITLE
Revert change to cse

### PR DESCRIPTION
**Pull request by bjodah**
This reverts changes to `cse_main` from gh-11232 which unfortunately degraded the CSE performance too much (see https://github.com/sympy/sympy_benchmarks/pull/38 & gh-12411).

Timings for `test_cse__performance` on my laptop:
- master: 935 s
- this PR: 0.1 s

EDIT: We should reopen gh-10228, when merging this.

### CODE REVIEW

**Comment by isuruf** [comment]
Can you add https://github.com/sympy/sympy/issues/12070 as a test here?

**Comment by bjodah** [comment]
@isuruf thanks, added it in the form of counting operations, including assignments. (6 vs 11 operations).
Is that what you had in mind?

**Comment by isuruf** [comment]
Thanks. that's perfect.

File: sympy/simplify/tests/test_cse.py [hunk 1]
```diff
@@ -175,17 +180,9 @@ def test_non_commutative_order():
     assert cse(l) == ([(x0, B+C)], [x0, A*x0])
 
 
-def test_issue_10228():
-    assert cse([x*y**2 + x*y]) == ([(x0, x*y)], [x0*y + x0])
-    assert cse([x + y, 2*x + y]) == ([(x0, x + y)], [x0, x + x0])
```
Comment (smichr) [inline sympy/simplify/tests/test_cse.py]: Please leave the issue number in place so it's easy to find. And I think this simple Add-sub expression extraction could be left, too. So just the first and second lines of this old test function. The issue should be reopened, too. I don't suppose there is an "opens #10228" syntax, however :-)

**Comment by bjodah** [comment]
Thanks @smichr, upon merging this we should re-open that issue.
(was 5df9a53 what you had in mind?)

**Comment by asmeurer** [comment]
@smichr can this be merged?

**Comment by smichr** [comment]
Leave a comment

> @smichr can this be merged?

done

**Comment by bjodah** [comment]
Opened gh-10228

**Pull request merged**